flink-runtime/src/main/java/org/apache/flink/runtime/state/SharedStateRegistry.java

```diff
@@ -57,13 +63,30 @@ public interface SharedStateRegistry extends AutoCloseable {
      */
     StreamStateHandle registerReference(
             SharedStateRegistryKey registrationKey, StreamStateHandle state, long checkpointID);
+
     /**
      * Unregister state that is not referenced by the given checkpoint ID or any newer.
      *
      * @param lowestCheckpointID which is still valid
      */
+    @VisibleForTesting
     void unregisterUnusedState(long lowestCheckpointID);
```

Comment:
The method is still used by production code, so the annotation is unnecessary.

Reply:
Delete it. It is used by `AbstractCompleteCheckpointStore#unregisterUnusedState()`, and  `AbstractCompleteCheckpointStore#unregisterUnusedState()` is used by `EmbeddedCompletedCheckpointStore#removeOldestCheckpoint()`, which is used for testing.